### ISSUE TITLE
Integrate embeddings computation into metrics logging

### ISSUE THREAD

**Issue by drigobon** [niznik-dev/predicting-zygosity#54]
# Describe the Feature

Add functionality to our fine-tuning training loop to save hidden state embeddings of input data. 

## Related Issues

Current metrics are logged in `_loss_step`, unclear if this is being run during validation or training steps.

# Acceptance Criteria
- [ ] Saves (csv? json?) embeddings files for comparison and analysis later.
- [ ] Expand config to include option of not saving embeddings matrices
- [ ] Ensure true labels / targets are properly associated with corresponding embeddings 
- [ ] Differentiate between training and validation embeddings?
